build.gradle.kts

```diff
@@ -1,33 +1,193 @@
+import org.springframework.boot.gradle.tasks.bundling.BootBuildImage
 
 plugins {
+    id("org.springframework.boot") version "2.7.7"
+    id("io.spring.dependency-management") version "1.1.0"
     kotlin("jvm") version "1.7.21"
+    kotlin("kapt") version "1.7.21"
+    kotlin("plugin.spring") version "1.7.21"
+    kotlin("plugin.jpa") version "1.7.21"
+    kotlin("plugin.allopen") version "1.7.21"
+    id("org.jlleitschuh.gradle.ktlint") version "11.0.0"
+    id("org.jlleitschuh.gradle.ktlint-idea") version "11.0.0"
+    id("org.openapi.generator") version "6.2.1"
+    id("org.owasp.dependencycheck") version "7.4.3"
 }
 
 group = "uk.gov.dluhc"
 version = "latest"
 java.sourceCompatibility = JavaVersion.VERSION_17
 
+ext["snakeyaml.version"] = "1.33"
+extra["springCloudVersion"] = "2.4.2"
+ext["spring-security.version"] = "5.7.5" // Fixed CVE-2022-31690 and CVE-2022-31692 - https://spring.io/blog/2022/10/31/cve-2022-31690-privilege-escalation-in-spring-security-oauth2-client
+extra["awsSdkVersion"] = "2.18.9"
+
+allOpen {
+    annotations("javax.persistence.Entity", "javax.persistence.MappedSuperclass", "javax.persistence.Embedabble")
+}
+
 repositories {
     mavenCentral()
 }
 
+apply(plugin = "org.jlleitschuh.gradle.ktlint")
+apply(plugin = "org.openapi.generator")
+apply(plugin = "org.springframework.boot")
+apply(plugin = "io.spring.dependency-management")
 apply(plugin = "org.jetbrains.kotlin.jvm")
+apply(plugin = "org.jetbrains.kotlin.plugin.spring")
+apply(plugin = "org.jetbrains.kotlin.plugin.jpa")
+apply(plugin = "org.jetbrains.kotlin.plugin.allopen")
 
 dependencies {
     // framework
     implementation("org.jetbrains.kotlin:kotlin-reflect")
     implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+    implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
+    implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310")
+    implementation("io.github.microutils:kotlin-logging-jvm:3.0.4")
+    implementation("org.apache.commons:commons-lang3:3.12.0")
+    // api
+    implementation("org.springframework.boot:spring-boot-starter-actuator")
+    implementation("org.springframework.boot:spring-boot-starter-web")
+    implementation("org.springdoc:springdoc-openapi-ui:1.6.13")
+    implementation("org.springframework.boot:spring-boot-starter-validation")
+
+    // webclient
+    implementation("org.springframework:spring-webflux")
+    implementation("io.projectreactor.netty:reactor-netty-http")
+
+    // spring security
+    implementation("org.springframework.boot:spring-boot-starter-security")
+    implementation("org.springframework.boot:spring-boot-starter-oauth2-resource-server")
+
+    implementation("io.awspring.cloud:spring-cloud-starter-aws-messaging")
+    implementation("org.springframework.boot:spring-boot-starter-data-jpa")
+    implementation("org.liquibase:liquibase-core")
+    implementation("org.hibernate:hibernate-envers")
```

Comment:
I think for this new service we ought to depend on spring-data-envers which will being in hibernate-envers transitively